src/Data/Api.re

```diff
@@ -0,0 +1,175 @@
+let faunaUrl = "https://graphql.fauna.com/graphql";
+
+let fetchConfessionsRequestInit = {
+  let allConfessionsQueryPayload = Js.Dict.empty();
+
+  Js.Dict.set(
+    allConfessionsQueryPayload,
+    "query",
+    Js.Json.string(Queries.allConfessionsQuery),
+  );
+
+  Fetch.RequestInit.make(
+    ~method_=Post,
+    ~body=
+      Fetch.BodyInit.make(
+        Js.Json.stringify(Js.Json.object_(allConfessionsQueryPayload)),
+      ),
+    ~headers=
+      Fetch.HeadersInit.make({"Authorization": "Bearer " ++ Env.token}),
+    (),
+  );
+};
+
+let fetchConfessions = callback => {
+  Js.Promise.(
+    Fetch.fetchWithInit(faunaUrl, fetchConfessionsRequestInit)
+    |> then_(Fetch.Response.json)
+    |> then_(json =>
+         json
+         |> ConfessionData.Decode.decodeGetRecentConfessionsResponse
+         |> (
+           decodedResponse =>
+             callback(
+               Some(decodedResponse.data.allConfessions.allConfessionsArray),
+             )
+             |> resolve
+         )
+       )
+    |> catch(_error => callback(None) |> resolve)
+    |> ignore
+  );
+};
+
+let fetchConfessionWithCommentsRequestInit = id => {
```

Comment:
API wrappers are one of the few places I personally feel is worth doing method level comments on. Ideally the contract shouldn't change super often and it's pretty nice as a dev coming onto a project, or as one reviewing code to know what the expected response of an API endpoint would be.